#! /bin/sh -e
## debian-libedit.dpatch by Mark Purcell <msp@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch asterisk to use installed libedit
## Build

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@

--- asterisk-0.9.1+1.0RC2.orig/Makefile
+++ asterisk-0.9.1+1.0RC2/Makefile
@@ -256,7 +260,7 @@
 		exit 1; \
 	fi
 
-asterisk: editline/libedit.a db1-ast/libdb1.a stdtime/libtime.a $(OBJS)
+asterisk: db1-ast/libdb1.a stdtime/libtime.a $(OBJS)
 	$(CC) $(DEBUG) -o asterisk $(ASTLINK) $(OBJS) $(LIBEDIT) db1-ast/libdb1.a stdtime/libtime.a $(LIBS)
 
 muted: muted.o
--- asterisk-0.9.1+1.0RC2.orig/cli.c
+++ asterisk-0.9.1+1.0RC2/cli.c
@@ -27,7 +27,7 @@
 #include <signal.h>
 #include <string.h>
 /* For rl_filename_completion */
-#include "editline/readline/readline.h"
+#include "readline/readline.h"
 /* For module directory */
 #include "asterisk.h"
 #include "build.h"
